codegen/transformer.go

```diff
@@ -0,0 +1,195 @@
+package codegen
+
+import (
+	"fmt"
+	"strings"
+
+	"goa.design/goa/expr"
+)
+
+type (
+	// Transformer transforms a source attribute to a target attribute.
```

Comment:
What is a transformation? also fix the English :)
Please explain carefully what this interface represents, how it's used etc. as it's a pretty fundamental one. Also it's a big interface which isn't great and a bit of a code smell - would be good to review and understand the need for it at a higher level.